game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyLogin.java

```diff
@@ -147,21 +136,40 @@ private void showError(final String title, final String message) {
     }
   }
 
-  private IMessenger createAccount(final String username, final String password, final String email)
-      throws IOException {
-    return new ClientMessenger(
-        lobbyServerProperties.getHost(),
-        lobbyServerProperties.getPort(),
-        username,
-        MacFinder.getHashedMacAddress(),
-        challenge -> {
-          final Map<String, String> response = new HashMap<>();
-          response.put(LobbyLoginResponseKeys.REGISTER_NEW_USER, Boolean.TRUE.toString());
-          response.put(LobbyLoginResponseKeys.EMAIL, email);
-          response.putAll(RsaAuthenticator.newResponse(challenge, password));
-          response.put(
-              LobbyLoginResponseKeys.LOBBY_VERSION, LobbyConstants.LOBBY_VERSION.toString());
-          return response;
-        });
+  private void forgotPassword() {
+    final ForgotPasswordPanel forgotPasswordPanel = ForgotPasswordPanel.newForgotPasswordPanel();
+    final ForgotPasswordPanel.ReturnValue returnValue = forgotPasswordPanel.show(parentWindow);
+    switch (returnValue) {
+      case OK:
+        forgotPassword(forgotPasswordPanel);
+        return;
+      case CANCEL:
```

Comment:
I get why you're using enums at this point, but I feel like just having a simple boolean would simplify this code a lot.

Reply:
In part, the account creation panel uses enums, so I want to try and stay consistent with that implementation. I'm not sure if a boolean return value is actually that much clear or not. Sure it's a few lines less code, but less explicit. I think perhaps something of a wash either way, in which case the consistency with the copy/pasted implementation wins the argument for me.

Reply:
What about checking if the return value is == to OK ?
This way you get to keep the enum and I get the simplicity of a simple if statement ^^

Reply:
At that point the enum does not make sense as you mentioned. The switch statement does not bother me.

The following does not read well:
```
if(forgotPasswordPanel.show(parentWindow)) {
:
:
}
```

We'd have to refactor the `show` method to be called something else. That would then break the symmetry with the "peer" panel classes. The ROI here I do not think is favorable here. Given that I don't mind the switch statement at all, for me the ROI is negative.

Reply:
I was actually thiking of this:
```java
if (forgotPasswordPanel.show(parentWindow) == ReturnValue.OK) {

}
```

Reply:
That is not too bad, I like the suggestion. I scanned the class, and we have this:
```
  private @Nullable LobbyClient loginToServer() {
    final LoginPanel loginPanel = new LoginPanel(LobbyLoginPreferences.load());
    final LoginPanel.ReturnValue returnValue = loginPanel.show(parentWindow);
    switch (returnValue) {
      case LOGON:
        return login(loginPanel);
      case CANCEL:
        return null;
      case CREATE_ACCOUNT:
        return createAccount();
      case FORGOT_PASSWORD:
        forgotPassword();
        return null;
      default:
        throw new AssertionError("unknown login panel return value: " + returnValue);
    }
  }
```

The other 3 switch statements could be simplified, but for now they are consistent with the first switch block. That makes me think it's better to not mix the patterns.